streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -1087,6 +1129,17 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
         );
     }
 
+    @SuppressWarnings("deprecation") // continuing to support Windows#maintainMs/segmentInterval in fallback mode
+    private static <K, V> StoreBuilder<WindowStore<K, V>> joinWindowStoreBuilderFromSupplier(final WindowBytesStoreSupplier storeSupplier,
+                                                                                             final Serde<K> keySerde,
+                                                                                             final Serde<V> valueSerde) {
+        return Stores.windowStoreBuilder(
+            storeSupplier,
+            keySerde,
+            valueSerde
+        );
+    }
+
     private class KStreamImplJoin {
```

Comment:
To reduce the complexity of `KStreamImpl` (and to avoid the checkstyle exception), could we move this nested class into its own file?

Reply:
ack